metamorph/src/main/java/org/metafacture/metamorph/maps/FileMap.java

```diff
@@ -29,21 +28,24 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.regex.Pattern;
+import java.util.Set;
 
 /**
- * Provides a {@link Map} based on a file. The file is supposed to be UTF-8
- * encoded. The default separator is {@code \t}. <strong>Important:</strong>
- * Lines that are not split in two parts by the separator are ignored!
+ * Provides an unmodifiable {@link Map} based on a file. The file is supposed to
+ * be UTF-8 encoded. The default separator is {@code \t}.
+ * <strong>Important:</strong> Lines that are not split in two parts by the
+ * separator are ignored!
  *
  * @author Markus Michael Geipel
  */
-public final class FileMap extends AbstractReadOnlyMap<String, String> {
+public final class FileMap extends HashMap<String, String> {
```

Comment:
Why are you de-`final`-izing `AbstractReadOnlyMap` if you're no longer inheriting from it?